### PR TITLE
Respect @property annotations on Entity classes

### LINKED ISSUE #172: Typehinting entity fields doesnt work due to overzealous `EntityFieldsViaMagicReflectionExtension`
In an attempt to suggest types to PHPStan, I found adding a `@property` hint to the class did not work when `phpstan-drupal` 's `PHPStan\Reflection\EntityFieldsViaMagicReflectionExtension` was enabled in extension.neon.

```php
/**
 * ....
 *
 * @property \Drupal\Core\Field\EntityReferenceFieldItemList $myField
 */
class MyEntity extends EntityBase {

  /**
   * ...
   */
  public function getParent(): array {
    return $this->myField->referencedEntities();
  }
```

Above code results in:

> Call to an undefined method Drupal\Core\Field\FieldItemListInterface::referencedEntities().

PHPStan thinks `myField` is a `FieldItemListInterface`. If I disable phpstan-drupal PHPStan does not complain.

A `@property` should probably take priority.

### PR DESCRIPTION
**Pull request by jacktonkin**
Fixes #172 and possibly another issue, since I'm sure I stole it from somewhere?

Updates EntityFieldsViaMagicReflectionExtensionTest to test annotated properties are skipped in the same way as explicit properties, adding an annotation to ReflectionEntityTest fixture.

No integration test to make sure annotations are actually respected, not sure if this is needed, or how to implement?

### CODE REVIEW

**Comment by jacktonkin** [comment]
Added a test for the annotated property.

**Review by mglaman — COMMENTED** [review]
Thanks for reporting this! I will probably merge as is and create an issue for my concern on leveraging the new Type system.

These are some of the oldest parts of phpstan-drupal and PHPStan changed a lot since they were added

**Comment by mglaman** [comment]
Also: don't worry about needing to expand the integration tests. Those are mostly to check if we broke autoloading somehow. And various contrib that had issues.

File: src/Reflection/EntityFieldsViaMagicReflectionExtension.php [hunk 1]
```diff
@@ -20,7 +20,7 @@ class EntityFieldsViaMagicReflectionExtension implements PropertiesClassReflecti
 
     public function hasProperty(ClassReflection $classReflection, string $propertyName): bool
     {
-        if ($classReflection->hasNativeProperty($propertyName)) {
+        if ($classReflection->hasNativeProperty($propertyName) || array_key_exists($propertyName, $classReflection->getPropertyTags())) {
```
[the remark this answers is not in the document]
Comment (mglaman) [reply src/Reflection/EntityFieldsViaMagicReflectionExtension.php:23]: We cannot use the type system, as `ObjectType::hasProperty` calls this reflection extension and we get an endless loop.

**Comment by mglaman** [comment]
I made a note as to _why_ this PR was needed and opened a discussion on PHPStan to see if we could improve our reflection extensions https://github.com/phpstan/phpstan/discussions/6207

**Pull request merged**